volumina/utility/cache.py

```diff
@@ -0,0 +1,119 @@
+import abc
+import sys
+import threading
+from collections import OrderedDict
+from typing import NamedTuple, Dict, Any, Type, TypeVar, Callable
+
+import numpy as np
+
+
+T = TypeVar("T")
+_256_MB = 256 * 1024 * 1024
+
+
+class CacheABC(abc.ABC):
```

Comment:
I touch method is a good idea, but I would like to avoid adding `__iter__` and ` __len__` as it's unnecessary for cache interface and we can get fun errors such as `RuntimeError: xxx size changed during iteration`

Reply:
could be circumvented with locks, probably

Reply:
Or just by copying the keyset. Also, iterating (and, therefore, getting the size) seems important: otherwise there is no way to find out the contents of a cache. Caches are just mappings anyway, so I would argue it's very Pythonic to make it behave like a dictionary.